include/deal.II/matrix_free/matrix_free.templates.h

```diff
@@ -337,12 +337,46 @@ MatrixFree<dim, Number, VectorizedArrayType>::internal_reinit(
             Utilities::MPI::this_mpi_process(task_info.communicator);
           task_info.n_procs =
             Utilities::MPI::n_mpi_processes(task_info.communicator);
+
+          Assert(additional_data.communicator_sm == MPI_COMM_SELF,
+                 ExcNotImplemented());
+
+#ifndef DEAL_II_WITH_MPI
+          Assert(false, ExcInternalError());
+#else
```

Comment:
Do we really need this assert, given that 15 lines above we do `job_supports_mpi()`?
```suggestion
#ifdef DEAL_II_WITH_MPI
```